util_windows.go

```diff
@@ -0,0 +1,166 @@
+package gvm
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/andrewkroh/gvm/common"
+	"github.com/pkg/errors"
+)
+
+func homeDir() (string, error) {
+	var homeDir string
+	if runtime.GOOS == "windows" {
+		homeDir = os.Getenv("USERPROFILE")
+	} else {
+		homeDir = os.Getenv("HOME")
+	}
+
+	if _, err := os.Stat(homeDir); err != nil {
+		return "", errors.Wrap(err, "failed to access home dir")
+	}
+
+	return homeDir, nil
+}
+
+func extractTo(to, file string) (string, error) {
+	tmpDir := to + ".tmp"
+	if err := os.Mkdir(tmpDir, 0755); err != nil {
+		return "", err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	if err := common.Extract(file, tmpDir); err != nil {
+		return "", err
+	}
+
+	// Move into the final location.
+	if err := copyDirectory(filepath.Join(tmpDir, "go"), to); err != nil {
+		return "", err
+	}
+	return to, nil
+}
+
+func existsDir(dir string) (bool, error) {
+	_, err := os.Stat(dir)
+	if err == nil {
+		return true, nil
+	}
+
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+func writeJsonFile(filename string, value interface{}) error {
+	contents, err := json.Marshal(value)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(filename, contents, 0644)
+}
+
+func readJsonFile(filename string, to interface{}) error {
+	contents, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(contents, to)
+}
+
+func copyDirectory(scrDir, dest string) error {
+	entries, err := ioutil.ReadDir(scrDir)
+	if err != nil {
+		return err
+	}
+	for _, entry := range entries {
+		sourcePath := filepath.Join(scrDir, entry.Name())
+		destPath := filepath.Join(dest, entry.Name())
+
+		fileInfo, err := os.Stat(sourcePath)
+		if err != nil {
+			return err
+		}
+
+		switch fileInfo.Mode() & os.ModeType {
+		case os.ModeDir:
+			if err := createIfNotExists(destPath, 0755); err != nil {
+				return err
+			}
+			if err := copyDirectory(sourcePath, destPath); err != nil {
+				return err
+			}
+		case os.ModeSymlink:
+			if err := copySymLink(sourcePath, destPath); err != nil {
+				return err
+			}
+		default:
+			if err := copyFile(sourcePath, destPath); err != nil {
+				return err
+			}
+		}
+
+		isSymlink := entry.Mode()&os.ModeSymlink != 0
+		if !isSymlink {
+			if err := os.Chmod(destPath, entry.Mode()); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func copyFile(srcFile, dstFile string) error {
```

Comment:
The functions below here appear to be duplicated for util_linux.go and util_windows.go. Rather than duplicating them put them into util.go.